Sources/Transport/Streams/Socks+Stream.swift

```diff
@@ -91,94 +99,53 @@ public final class TCPServerStream: TCPProgramStream, ServerStream {
     }
 
     public func accept() throws -> Stream {
-        let next = try stream.accept()
+
         switch securityLayer {
         case .none:
-            return next
-        case .tls:
-            let secure = try next.makeSecret(mode: .server)
-            try secure.stream.accept()
+            return try stream.accept()
+        case .tls(let provided):
+            let config: Config
+            if let c = provided {
+                config = c
+            } else {
+                let context = try Context(mode: .server)
+                config = try Config(context: context)
+            }
+
+            let secure = try TLS.Socket(config: config, socket: stream)
+            try secure.accept()
             return secure
         }
     }
 }
 
-import TLS
-import SocksCore
-import libc
-
-public final class TLSSocket<Socket: SocksCore.TCPInternetSocket> {
-    public let stream: TLS.Stream
-    public let socket: Socket
-
-    public init(stream: TLS.Stream, socket: Socket) {
-        self.stream = stream
-        self.socket = socket
-    }
-
-    public convenience init(
-        mode: TLS.Mode,
-        socket: Socket,
-        certificates: TLS.Certificates = .none
-    ) throws {
-        let stream = try TLS.Stream(
-            mode: mode,
-            socket: socket.descriptor,
-            certificates: certificates
-        )
-        self.init(stream: stream, socket: socket)
-    }
-}
-
-/**
-    Conform TLS Stream + Underlying socket
-    to a TLS Socket.
-*/
-extension TLSSocket: Stream {
-    public func close() throws {
-        try stream.close()
-        try socket.close()
-    }
-
-    public var closed: Bool {
-        return socket.closed
-    }
-
-    public var peerAddress: String {
-        return socket.peerAddress
-    }
-
+extension TLS.Socket: Stream {
     public func setTimeout(_ timeout: Double) throws {
         try socket.setTimeout(timeout)
     }
 
-    public func send(_ bytes: Bytes) throws {
-        try stream.send(bytes)
-    }
-
-    public func receive(max: Int) throws -> Bytes {
-        return try stream.receive(max: max)
+    public var closed: Bool {
+        return socket.closed
     }
 
     public func flush() throws {
         try socket.flush()
     }
+
+    public var peerAddress: String {
+        return socket.peerAddress
+    }
 }
 
-extension TCPInternetSocket {
-    /**
-        Creates a new SSL Context and Secure Socket.
-        - parameter mode: Client or Server
-        - parameter certificates: SSL Certificates for the Server, use .none for Client
-     */
-    public func makeSecret(
-        mode: TLS.Mode = .client,
-        certificates: TLS.Certificates = .none
-    ) throws -> TLSSocket<TCPInternetSocket> {
-        return try TLSSocket(
-            mode: mode,
-            socket: self,
-            certificates: certificates
-        )
+extension SecurityLayer: Equatable {
```

Comment:
I thought we took out equatable, was that a different PR? I think we should let people deal w/ this one on their own, maybe a convenience like:

```
var isNone: Bool
var isTLS: Bool
```